File: Mage/CoreData/ObservationFavorite.m
//
//  ObservationFavorite+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by William Newman on 9/20/16.
//  Copyright © 2016 National Geospatial-Intelligence Agency. All rights reserved.
//

#import "ObservationFavorite.h"

@implementation ObservationFavorite

+ (ObservationFavorite *) favoriteForUserId: (NSString *) userId inManagedObjectContext:(NSManagedObjectContext *) context {
    ObservationFavorite *favorite = [ObservationFavorite MR_createEntityInContext:context];
    
    favorite.dirty = NO;
    favorite.favorite = YES;
    favorite.userId = userId;
    
    return favorite;
}

@end

File: Mage/CoreData/ObservationFavorite.swift
//
//  ObservationFavorite+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by William Newman on 9/20/16.
//  Copyright © 2016 National Geospatial-Intelligence Agency. All rights reserved.
//

import Foundation
import CoreData

@objc public class ObservationFavorite: NSManagedObject {
    @objc public static func favorite(userId: String, context: NSManagedObjectContext) -> ObservationFavorite? {
        let favorite = ObservationFavorite.mr_createEntity(in: context);
        favorite?.dirty = false
        favorite?.favorite = true;
        favorite?.userId = userId;
        return favorite;
    }
}
